packages/flutter_tools/lib/src/resident_web_runner.dart

```diff
@@ -130,76 +281,287 @@ class ResidentWebRunner extends ResidentRunner {
+        'Content-Type': 'text/javascript',
+      });
+    } else if (request.url.path.contains('.bootstrap.js')) {
+      final File file = fs.file(fs.path.join(
+        generated,
+        'lib',
+        'main_web_entrypoint.dart.bootstrap.js',
+      ));
+      return Response.ok(file.readAsBytesSync(), headers: <String, String>{
+        'Content-Type': 'text/javascript',
+      });
+    } else if (request.url.path.contains('dart_sdk')) {
+      final File file = fs.file(fs.path.join(
+        artifacts.getArtifactPath(Artifact.flutterWebSdk),
+        'kernel',
+        'amd',
+        'dart_sdk.js',
+      ));
+      return Response.ok(file.readAsBytesSync(), headers: <String, String>{
+        'Content-Type': 'text/javascript',
+      });
+    } else if (request.url.path.contains('main_web_entrypoint.digests')) {
+      final File file = fs.file(fs.path.join(
+        generated,
+        'lib',
+        'main_web_entrypoint.digests',
+      ));
+      return Response.ok(file.readAsBytesSync());
+    } else if (request.url.path.contains('assets')) {
+      final String assetPath = request.url.path.replaceFirst('assets/', '');
+      final File file = fs.file(fs.path.join(getAssetBuildDirectory(), assetPath));
+      return Response.ok(file.readAsBytesSync());
+    }
+    return Response.notFound('');
+  }
+
   @override
   Future<OperationResult> restart({
     bool fullRestart = false,
     bool pauseAfterRestart = false,
     String reason,
     bool benchmarkMode = false,
   }) async {
+    if (!fullRestart) {
+      return OperationResult(1, 'hotReload not supported');
+    }
     final Stopwatch timer = Stopwatch()..start();
     final Status status = logger.startProgress(
       'Performing hot restart...',
       timeout: timeoutConfiguration.fastOperation,
       progressId: 'hot.restart',
     );
-    OperationResult result = OperationResult.ok;
-    try {
-      final List<Uri> invalidatedSources = ProjectFileInvalidator.findInvalidated(
-        lastCompiled: _lastCompiled,
-        urisToMonitor: <Uri>[
-          for (FileSystemEntity entity in flutterProject.directory
-              .childDirectory('lib')
-              .listSync(recursive: true))
-            if (entity is File && entity.path.endsWith('.dart')) entity.uri
-        ], // Add new class to track this for web.
-        packagesPath: PackageMap.globalPackagesPath,
-      );
-      await webCompilationProxy.invalidate(inputs: invalidatedSources);
-      await _connection.sendCommand('Page.reload');
-      await Future<void>.delayed(const Duration(milliseconds: 150));
-    } catch (err) {
-      result = OperationResult(1, err.toString());
-    } finally {
-      printStatus('Restarted application in ${getElapsedAsMilliseconds(timer.elapsed)}.');
-      status.cancel();
-    }
-    return result;
+    final vmservice.Response reloadResponse = await _vmService.callServiceExtension('hotRestart');
```

Comment:
Currently the daemon is running eagerly and this is triggering the reload
- but this is where we'd like to be passing the results to the daemon and then triggering the hot restart.